presto-native-execution/CMakeLists.txt

```diff
@@ -160,9 +159,18 @@ endif()
 include_directories(SYSTEM ${FOLLY_INCLUDE_DIRS})
 
 # Include third party header files
+
+# Find openssl@1.1 folder.
 find_path(OPT_OPENSSL_DIR NAMES opt/openssl@1.1)
+if(${OPT_OPENSSL_DIR} STREQUAL "OPT_OPENSSL_DIR-NOTFOUND")
+  find_path(OPT_OPENSSL_DIR NAMES ../opt/openssl@1.1)
+  if(NOT ${OPT_OPENSSL_DIR} STREQUAL "OPT_OPENSSL_DIR-NOTFOUND")
+    set(OPT_OPENSSL_DIR "${OPT_OPENSSL_DIR}/..")
+  endif()
+endif()
 set(OPENSSL_ROOT_DIR "${OPT_OPENSSL_DIR}/opt/openssl@1.1")
 find_package(OpenSSL REQUIRED)
+include_directories(SYSTEM ${OPENSSL_ROOT_DIR}/include)
```

Comment:
Shouldnt find package add the openssl include dirs ?

Reply:
Well... maybe. :)
But I think "include_directories" is just adding -I or -isystem argument for compiler and might not affect find_package() at all.
Somehow it works this way too.
I can swap them.

Reply:
include_directories will explicitly add this path for all translation units irrespective of whether it uses it - But if find package succeeds then only the targets that use it (directly or transitively) would have it included. Does your build fail without it ?

Reply:
@kgpai 
I haven't run build w/o "find_package". It was there and I assume it is needed.
My build fails if I didn't add openssl into "include_directories" before adding "Boost_INCLUDE_DIRS".

Reply:
I reckon that what is happening is that this issue is masking some other problem - having to include_directories openssl@1.1 likely means that it was a) not getting openssl dependency, b) or this include masks it picking up openssl from other source (likely one @ version 3). I believe having to worry about order of cmake includes might be an antipattern.. .

Reply:
@kgpai 
Order is extremely important for systems with multiple versions of libraries.
If you only have one library, then order is not important.
One of the actual problems (from my perspective) here is that Boost_INCLUDE_DIRS end up pointing at the common place "/Users/homebrew/include", which has nothing to do with boost at all. Does this not trigger your interest?